[nvbugs/5345391] fix: chunked prefill + overlap scheduling

Funatiq opened the pull request:
## Description

The main problem in the overlap path is that 
`previous_batch_size = self.previous_batch.sample_state.scheduled_requests.batch_size` 
does not consider the chunked context requests which are separated from context requests in the overlap loop.

Removed the batch_size check so previous batch is always processed.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
test: Re-enable overlap scheduler in test_llm_api_pytorch::test_chunked_prefill

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #11016](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11016/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #11016](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11016/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.21/L0_MergeRequest_PR pipeline #168](https://nv/trt-llm-cicd/job/release-0.21/job/L0_MergeRequest_PR/168/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #11147](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11147/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.21/L0_MergeRequest_PR pipeline #185](https://nv/trt-llm-cicd/job/release-0.21/job/L0_MergeRequest_PR/185/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/release-0.21/L0_MergeRequest_PR/185/test-results/rerun_report.html) for details.

Funatiq committed:
fix: chunked prefill + overlap scheduling

- Don't skip previous_batch based on batch size since it doesn't include chunked requests.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #11179](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11179/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #11300](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11300/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.21/L0_MergeRequest_PR pipeline #195](https://nv/trt-llm-cicd/job/release-0.21/job/L0_MergeRequest_PR/195/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/release-0.21/L0_MergeRequest_PR/195/test-results/rerun_report.html) for details.
